libs/libc/stream/lib_hexdumpstream.c

```diff
@@ -0,0 +1,182 @@
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/compiler.h>
+#include <nuttx/streams.h>
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: nibble2hex
+ *
+ * Description:
+ *  Convert a binary nibble to a hexadecimal character.
+ *
+ ****************************************************************************/
+
+static char nibble2hex(unsigned char nibble)
+{
+  if (nibble < 10)
+    {
+      return '0' + nibble;
+    }
+  else
+    {
+      return 'A' + nibble - 10;
+    }
+}
+
+/****************************************************************************
+ * Name: bin2hex
+ ****************************************************************************/
+
+static size_t bin2hex(FAR const uint8_t *buf, size_t buflen,
+                      FAR char *hex, size_t hexlen)
+{
+  size_t i;
+
+  if (buflen > hexlen)
+    {
+      buflen = hexlen;
+    }
+
+  for (i = 0; i < buflen; i++)
+    {
+      hex[2 * i]     = nibble2hex(buf[i] >> 4);
+      hex[2 * i + 1] = nibble2hex(buf[i] & 0xf);
+    }
+
+  return buflen;
+}
+
+/****************************************************************************
+ * Name: hexdumpstream_flush
+ ****************************************************************************/
+
+static int hexdumpstream_flush(FAR struct lib_outstream_s *this)
+{
+  FAR struct lib_hexdumpstream_s *rthis = (FAR void *)this;
+
+  if (rthis->pending > 0)
+    {
+      rthis->buffer[rthis->pending] = '\n';
+      lib_stream_puts(rthis->backend, rthis->buffer, rthis->pending + 1);
+      rthis->pending = 0;
+    }
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: hexdumpstream_putc
+ ****************************************************************************/
+
+static void hexdumpstream_putc(FAR struct lib_outstream_s *this, int ch)
+{
+  FAR struct lib_hexdumpstream_s *rthis = (FAR void *)this;
+  int outlen = sizeof(rthis->buffer);
```

Comment:
change sizeof(rthis->buffer) to CONFIG_STREAM_HEXDUMP_BUFFER_SIZE

Reply:
Done